contracts/transfer/src/state.rs

```diff
@@ -302,6 +303,55 @@ impl TransferState {
         }
     }
 
+    /// Transfer funds from one contract's balance to another.
+    ///
+    /// Contracts can call the function and expect that if it succeeds the funds
+    /// are succesfully transferred to the contract they specify. Contracts
+    /// receiving funds are expected to expose the function specified by the
+    /// sender, which is called using a [`ReceiveFromContract`] as argument. It
+    /// is recommended that the receiving contract check that the call
+    /// originates from the transfer contract, and subsequently run any logic it
+    /// may wish to handle the transfer - including panicking, which will
+    /// effectively reject the transfer.
+    ///
+    /// # Panics
+    /// The function will panic if it is not being called by a contract *other*
+    /// than the transfer contract, if the call to the receiving contract fails,
```

Comment:
to me this reads that only the transfer contract can call this function. Maybe go for 
```suggestion
    /// The function will panic if it is not being called by a contract (or if it is called
    /// by the transfer contract itself), if the call to the receiving contract fails,
```

Reply:
I did struggle with phrasing this. Thank you!